apps/aecore/lib/aecore/chain/chainstate.ex

```diff
@@ -59,68 +57,18 @@ defmodule Aecore.Chain.Chainstate do
     end
   end
 
-  @spec apply_transaction_on_state(SignedTx.t(), Chainstate.t(), non_neg_integer()) ::
+  @spec apply_transaction_on_state(Chainstate.t(), non_neg_integer(), SignedTx.t()) ::
           Chainstate.t()
-  def apply_transaction_on_state(
-        %{data: %{sender: nil, payload: %{receiver: receiver}} = data, signature: nil},
-        %{accounts: accounts} = chainstate,
-        block_height
-      ) do
-    receiver_state = Account.get_account_state(accounts, receiver)
-    new_receiver_state = SignedTx.reward(data, receiver_state)
-    updated_receiver_state = %{new_receiver_state | last_updated: block_height}
-
-    new_accounts_state =
-      AccountStateTree.put(chainstate.accounts, data.payload.receiver, updated_receiver_state)
-
-    {:ok, Map.put(chainstate, :accounts, new_accounts_state)}
-  end
+  def apply_transaction_on_state(chainstate, block_height, tx) do
+    case SignedTx.validate(tx) do
+      :ok ->
+        SignedTx.process_chainstate(chainstate, block_height, tx)
```

Comment:
Why do you think it's better to hide the called functions in some nested calls, instead of having them explicit here as before?

Reply:
Because more abstraction is better.